tests/algorithms/test_algorithm_resumption.py

```diff
@@ -1,105 +1,106 @@
 # Copyright 2022 MosaicML Composer authors
 # SPDX-License-Identifier: Apache-2.0
 
+import copy
 import os
 import pathlib
-from typing import Optional
+from typing import Any, Callable, Dict
 
 import pytest
 import torch
-from torch.utils.data import DataLoader
+from torch.utils.data import DataLoader, Dataset
 
 from composer import Trainer
-from composer.algorithms import get_algorithm_registry
-from tests.algorithms.algorithm_settings import get_settings
+from composer.algorithms.factorize.factorize import Factorize
+from composer.algorithms.layer_freezing.layer_freezing import LayerFreezing
+from composer.algorithms.sam.sam import SAM
+from composer.algorithms.squeeze_excite.squeeze_excite import SqueezeExcite
+from composer.algorithms.stochastic_depth.stochastic_depth import StochasticDepth
+from composer.core.algorithm import Algorithm
+from composer.models.base import ComposerModel
+from tests.algorithms.algorithm_settings import get_algorithm_parametrization
 from tests.common import deep_compare, device
 
-ALGORITHMS = get_algorithm_registry().keys()
+
+@pytest.fixture
+def model(request):
+    # Copy the model, so other parameterization start with a fresh model
+    return copy.deepcopy(request.param)
 
 
 @pytest.mark.timeout(180)
 @device('gpu')
 @pytest.mark.parametrize(
-    "seed,save_interval,save_filename,resume_file,final_checkpoint",
+    "save_interval,save_filename,resume_file,final_checkpoint",
     [
-        [None, "1ep", "ep{epoch}-rank{rank}", "ep2-rank{rank}", "latest-rank{rank}"
-        ],  # test randomized seed saving and symlinking
-        [42, "1ep", "ep{epoch}-rank{rank}", "ep3-rank{rank}", "ep5-rank{rank}"],  # test save at epoch end
+        ["1ep", "ep{epoch}-rank{rank}", "ep2-rank{rank}", "latest-rank{rank}"],  # symlinking
+        ["1ep", "ep{epoch}-rank{rank}", "ep3-rank{rank}", "ep5-rank{rank}"],  # test save at epoch end
     ],
 )
-@pytest.mark.filterwarnings(r"ignore:.*Detected call of \`lr_schedule.*:UserWarning")  # Needed for SWA
-@pytest.mark.parametrize("algorithm", ALGORITHMS)
+@pytest.mark.parametrize("alg_cls,alg_kwargs,model,dataset", get_algorithm_parametrization(), indirect=['model'])
```

Comment:
Could we retain the option to select certain algorithms? This is useful for testing the resumption of individual algos during test time

Reply:
I would prefer the previous pattern of retrieval rather than have four bundled parameterizations in the unit tests. More modularized that way.

Reply:
Makes sense; I like this approach. I will rename `get_algorithm_parametrization` to `get_algs_and_marks`, then add back the retrieval functions for `get_alg_kwargs`, `get_alg_model`, and `get_alg_dataset` that individual tests can call.

Reply:
(Individual test files can filter the list returned by `get_algs_and_marks` before it is passed to `pytest.mark.parametrize` -- see the torchscript tests as an example)

Reply:
Refactored